docs/topics/whatsnew/whatsnew-eap.md

```diff
@@ -108,8 +251,203 @@ kotlin {
+This process was error-prone, especially for overloaded functions.
+
+Now, in Kotlin %kotlinEapVersion%, the [Kotlin Metadata JVM library](metadata-jvm.md) introduces support for reading annotations stored in Kotlin metadata.
+
+To make annotations available in the metadata for your compiled files, add the following compiler option:
+
+```kotlin
+-Xannotations-in-metadata
+```
+
+Alternatively, add it to the `compilerOptions {}` block of your Gradle build file:
+
+```kotlin
+// build.gradle.kts
+kotlin {
+    compilerOptions {
+        freeCompilerArgs.add("-Xannotations-in-metadata")
+    }
+}
+```
+
+With this option enabled, the Kotlin compiler writes annotations into metadata alongside the JVM bytecode, making them accessible to the `kotlin-metadata-jvm` library.
+
+The library provides the following APIs for accessing annotations:
+
+* `KmClass.annotations`
+* `KmFunction.annotations`
+* `KmProperty.annotations`
+* `KmConstructor.annotations`
+* `KmPropertyAccessorAttributes.annotations`
+* `KmValueParameter.annotations`
+* `KmFunction.extensionReceiverAnnotations`
+* `KmProperty.extensionReceiverAnnotations`
+* `KmProperty.backingFieldAnnotations`
+* `KmProperty.delegateFieldAnnotations`
+* `KmEnumEntry.annotations`
+
+These APIs are [Experimental](components-stability.md#stability-levels-explained).
+To opt in, use the `@OptIn(ExperimentalAnnotationsInMetadata::class)` annotation.
+
+Here's an example of reading annotations from Kotlin metadata:
+
+```kotlin
+@file:OptIn(ExperimentalAnnotationsInMetadata::class)
+
+import kotlin.metadata.ExperimentalAnnotationsInMetadata
+import kotlin.metadata.jvm.KotlinClassMetadata
+
+annotation class Label(val value: String)
+
+@Label("Message class")
+class Message
+
+fun main() {
+    val metadata = Message::class.java.getAnnotation(Metadata::class.java)
+    val kmClass = (KotlinClassMetadata.readStrict(metadata) as KotlinClassMetadata.Class).kmClass
+    println(kmClass.annotations)
+    // [@Label(value = StringValue("Message class"))]
+}
+```
+
+> If you use the `kotlin-metadata-jvm` library in your projects, we recommend testing and updating your code to support annotations.
+> Otherwise, when annotations in metadata become [enabled by default](https://youtrack.jetbrains.com/issue/KT-75736) in a future Kotlin version, your projects may produce invalid or incomplete metadata.
+>
+> If you experience any problems, please report them in our [issue tracker](https://youtrack.jetbrains.com/issue/KT-31857).
+>
+{style="warning"}
+
+## Kotlin/Native
+
+### Per-object memory allocation
+
+<primary-label ref="experimental-opt-in"/>
+
+Kotlin/Native’s [memory allocator](https://github.com/JetBrains/kotlin/blob/master/kotlin-native/runtime/src/alloc/custom/README.md) can now reserve memory on a per-object basis. In some cases, it may help you avoid strict memory limitations or high memory consumption on the application's startup.
```

Comment:
```suggestion
Kotlin/Native's [memory allocator](https://github.com/JetBrains/kotlin/blob/master/kotlin-native/runtime/src/alloc/custom/README.md) can now reserve memory on a per-object basis. In some cases, it may help you avoid strict memory limitations or high memory consumption on the application's startup.
```